### PR TITLE
Blog

### PR DESCRIPTION
**Pull request by tabbykatz**
## Overview

_First attempt at routing, very enjoyable_

## Feedback Requested

_I'm assuming I did not stumble upon the very best way to handle this._
 

- how can I improve?
- next steps?

### CODE REVIEW

**Comment by tabbykatz** [comment]
@Clark73 😛

File: server/blogRouter.mjs [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+blogRouter.get("/:id", async (request, response) => {
+  const post = await db.getPostById(request.params.id);
+  response.json(post);
+});
+
+blogRouter.use(express.json());
+blogRouter.post("/", async (request, response) => {
+  response.status(201).json(await db.addPost(request.body));
+});
+blogRouter.post("/:id", async (request, response) => {
```
Comment (kevin-scrib) [inline server/blogRouter.mjs:20]: I feel like this should be a POST /posts/:id/comments.    doing a POST /posts/:id makes me feel like I'm editing something about the post it self.

File: server/db.mjs [hunk 2]
```diff
@@ -3,11 +3,33 @@ import pgp from "pg-promise";
 
 const db = initDb();
 
-export const getTasks = () => db.any("SELECT * FROM tasks");
+export const getPostsWithComments = () =>
+  db.any(
+    "SELECT posts.*, JSON_AGG(comments) AS comments FROM posts LEFT JOIN comments ON posts.id = comments.post_id GROUP BY posts.id ORDER BY posts.id DESC",
```
Comment (kevin-scrib) [inline server/db.mjs:8]: nice

File: app/src/App.js [hunk 3]
```diff
@@ -1,35 +1,110 @@
...
+  return (
+    <>
+      <nav>
+        <Link to="/">Blog</Link> | <Link to="add-post">Write Post</Link>
+      </nav>
+      <main>
+        <Routes>
+          <Route path="/add-post" element={<WritePost />} />
+          <Route path="/" element={<Blog />} />
+          <Route path="/post/:id/:slug" element={<Post />} />
```
Comment (kevin-scrib) [inline app/src/App.js:23]: where do you use the :/slug? I can not find any use of it

File: app/src/App.js [hunk 4]
```diff
@@ -1,35 +1,110 @@
...
+    </>
+  );
+};
+
+const WritePost = () => {
+  const [post, setPost] = React.useState("Markdown &hearts;");
+  const [title, setTitle] = React.useState("");
+  let navigate = useNavigate();
+
+  const addPost = (e) => {
```
Comment (kevin-scrib) [inline app/src/App.js:35]: what is e? maybe this could be named a bit better

File: app/src/App.js (hunk 3 above)
Comment (kevin-scrib) [reply app/src/App.js:23]: are you trying to make it so you can load a post by its title?

**Comment by kevin-scrib** [comment]
Great work on this, looks very clean and easy to follow

File: server/blogRouter.mjs (hunk 1 above)
Comment (tabbykatz) [reply server/blogRouter.mjs:20]: I trust your feeling & I'd love to talk more about this. Should I add /comments just for descriptive purposes?

File: app/src/App.js (hunk 3 above)
Comment (tabbykatz) [reply app/src/App.js:23]: Yes, I was. I will submit a new PR and you can see my solution.

File: app/src/App.js (hunk 4 above)
Comment (tabbykatz) [reply app/src/App.js:35]: It's the event. You're right. Thanks!

**Pull request closed without merging**